src/paint.js

```diff
@@ -598,19 +608,35 @@ function setupPaint({ $, qlik }) {
               return colors[i];
             })
             .style("opacity", tileOpacity)
-            .on("mouseenter", function (d, i) {
-              d3.selectAll('[fill="' + colors[i] + '"]')
+            .on("mouseenter", function (dataObject, index) {
+              d3.selectAll('[fill="' + colors[index] + '"]')
                 .classed({
                   "bordered": false,
                   "borderedHover": true
                 });
             })
-            .on("mouseleave", function (d, i) {
-              d3.selectAll('[fill="' + colors[i] + '"]')
+            .on("mouseleave", function (dataObject, index) {
+              d3.selectAll('[fill="' + colors[index] + '"]')
                 .classed({
                   "bordered": true,
                   "borderedHover": false
                 });
+            })
+            .on('click', function (d , i){
```

Comment:
+1 on giving `d` and `i` more descriptive names. Also I would add that from usability point of view we need to enable the click event on the whole legend element, not just the bar. The text should also be clickable as it has more space than the bar.